src/finch/algebra/algebra.py

```diff
@@ -210,6 +211,21 @@ def _return_type_closure(a, b):
         register_property(T, meth, "return_type", _return_type_reflexive(meth))
         register_property(T, rmeth, "return_type", _return_type_reflexive(rmeth))
 
+register_property(
+    min,
+    "__call__",
+    "return_type",
+    lambda op, a, b: query_property(a, "__add__", "return_type", b),
+)
+register_property(
+    max,
+    "__call__",
+    "return_type",
+    lambda op, a, b: query_property(a, "__add__", "return_type", b),
```

Comment:
why not just add min and max to the list of reflexive operators?

Reply:
Defining `return_type` of max differently than the actual return type of max is incorrect.